tests/test_schema.py

```diff
@@ -593,26 +594,28 @@ def test_patchset_fail(datadir, patchset_file):
 
 
 @pytest.fixture
-def refresh_pyhf(monkeypatch):
-    global pyhf
-    modules_to_clear = [name for name in sys.modules if name.split('.')[0] == 'pyhf']
-    for module_name in modules_to_clear:
-        monkeypatch.delitem(sys.modules, module_name)
-    pyhf = importlib.import_module(pyhf.__name__)
-    return pyhf
+def monkeypatched_sys_modules():
+    with unittest.mock.patch.dict(sys.modules):
+        yield sys.modules
 
 
 def test_defs_always_cached(
     socket_disabled,  # noqa: F811
-    refresh_pyhf,  # ensure there is not a pre-existing cache hiding the issue
+    monkeypatched_sys_modules,
 ):
     """
     Schema definitions should always be loaded from the local files and cached at first import.
 
-    Otherwise using pyhf in contexts where the jsonschema.RefResolver cannot lookup the definition by the schema-id,
-    it will crash (e.g. a cluster node without network access).
+    Otherwise pyhf will crash in contexts where the jsonschema.RefResolver cannot lookup the definition by the schema-id
+    (e.g. a cluster node without network access).
     """
-    pyhf = refresh_pyhf
+
+    global pyhf
+    modules_to_clear = [name for name in sys.modules if name.split('.')[0] == 'pyhf']
+    for module_name in modules_to_clear:
+        del monkeypatched_sys_modules[module_name]
+    pyhf = importlib.import_module(pyhf.__name__)
```

Comment:
just `importlib.import_module('pyhf')` as we won't be renaming it anytime soon...